run-basic-batch-operators-test.sh

```diff
@@ -120,28 +139,32 @@ withRetry() {
 			# If the command succeeded, break out of the loop
 			# shellcheck disable=SC2031
 			echo "$stdout"
-
-			echo "command: $*" >&2
-			echo "stderr: $stderr" >&2
-			echo "stdout: $stdout" >&2
-			echo "status: $status" >&2
+			{
+				echo "command: $*"
+				echo "stderr: $stderr"
+				echo "stdout: $stdout"
+				echo "status: $status"
+			} >>"$LOG_FILE_PATH"
 			return 0
 		fi
 		# If the command failed, increment the retry counter
 		retries=$((retries + 1))
-		# shellcheck disable=SC2031
-		echo "command: $*" >&2
-		# shellcheck disable=SC2031
-		echo "stderr: $stderr" >&2
-		# shellcheck disable=SC2031
-		echo "stdout: $stdout" >&2
-		# shellcheck disable=SC2031
-		echo "status: $status" >&2
-		echo "Retry $retries/$maxRetries: Waiting for a few seconds before the next attempt..." >&2
+		{
+			# shellcheck disable=SC2031
+			echo "command: $*"
+			# shellcheck disable=SC2031
+			echo "stderr: $stderr"
+			# shellcheck disable=SC2031
+			echo "stdout: $stdout"
+			# shellcheck disable=SC2031
+			echo "status: $status"
+		} >>"$LOG_FILE_PATH"
+		echoColor "$GREY" "Waiting for a few seconds before the next attempt..."
+		echo "Retry $retries/$maxRetries: Waiting for a few seconds before the next attempt..." >>"$LOG_FILE_PATH"
```

Comment:
If `echoColor` already redirects the message to the file, what is the point of doing it twice?

Reply:
fixed